### ISSUE TITLE
Custom logger's log_metrics method is never called

### ISSUE THREAD

**Issue by ymfa** [PyTorchLightning/pytorch-lightning#7533]
## 🐛 Bug

I was following the example given on https://pytorch-lightning.readthedocs.io/en/latest/extensions/logging.html#make-a-custom-logger to create a custom logger by extending LightningLoggerBase. When I train my model using the custom logger, the log_metrics() method of the custom logger is never reached, and hence no metrics are logged.

As I'm new to pytorch-lightning, I suspect this might not be a genuine bug, but instead it might mean the documentation needs updating.

### To Reproduce

Here is a dummy logger that reproduces this problem. It is mostly identical to the given example.

```
import datasets
import pytorch_lightning as pl
import torch
from pytorch_lightning.loggers import LightningLoggerBase
from pytorch_lightning.loggers.base import rank_zero_experiment
from pytorch_lightning.utilities import rank_zero_only
from torch import nn


class MyLogger(LightningLoggerBase):

    @property
    def name(self):
        return 'MyLogger'

    @property
    @rank_zero_experiment
    def experiment(self):
        return '0.1'  # I don't understand what an 'experiment' is

    @property
    def version(self):
        return '0.1'

    @rank_zero_only
    def log_hyperparams(self, params):
        print('log hyperparams')

    @rank_zero_only
    def log_metrics(self, metrics, step):
        print('log metrics')

    @rank_zero_only
    def save(self):
        super().save()

    @rank_zero_only
    def finalize(self, status):
        pass
```

This is how I used the logger:

```
class DummyModel(pl.LightningModule):

    def __init__(self):
        super().__init__()
        self.encoder = nn.Linear(2, 1)

    def training_step(self, *args, **kwargs):
        pass

    def validation_step(self, *args, **kwargs):
        self.log('some_metric', 101)

    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
        return optimizer


trainer = pl.Trainer(logger=MyLogger())
model = DummyModel()
dataset = datasets.load_dataset('gem', 'xsum')
trainer.fit(model, dataset['train'], dataset['validation'])
```

### Expected behavior

When I train a model using this dummy logger, I expect the message 'log metrics' being printed each time metrics are to be logged. However, I only see 'log hyperparams' being printed, but not 'log metrics'. Apparently, `log_metrics()` is never called.

Note that my dummy model doesn't implement `training_step()`, but it does call `self.log()` in `validation_step()`. So I expect the logger to be used during the "sanity check" validation (which is before the actual training).

### Environment

 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): macOS
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Python version: 3.7.4
 - CUDA/cuDNN version: n/a
 - GPU models and configuration: n/a

**Comment by rohitgr7** [comment]
I tried your example and it's working fine. Can you try it with the latest release?

**Comment by ymfa** [comment]
> I tried your example and it's working fine. Can you try it with the latest release?

I'm on pytorch-lightning-1.3.1, which I think is the latest release. Did you see the message 'log metrics' printed out in console?

**Comment by rohitgr7** [comment]
sanity check isn't disabled but logging is disabled during sanity check that's why you are not seeing the print there.

**Comment by ymfa** [comment]
> sanity check isn't disabled but logging is disabled during sanity check that's why you are not seeing the print there.

Yes, thanks. I can make the minimal example work now. My real model, however, is still not logging anything and I'm in the process of debugging. I'll close this issue later when I get down to the real cause of the problem.

**Comment by ymfa** [comment]
It turns out that the issue was simply that the first logging step occurs too late. Due to the following reasons, logging only starts after completing the first complete validation:
1. Logging statements have no effect during the sanity check validation.
2. A `self.log` statement in `validation_step` is not `on_step=True` by default.
3. The `log_every_n_steps` parameter of the Trainer has a large default value (50), and is further complicated by my use of `accumulate_grad_batches`. It means I have to wait `log_every_n_steps * accumulate_grad_batches` training steps to see the first logged loss, and this happens to be more than one epoch. I think the documentation could be clearer on this.
